src/player.js

```diff
@@ -1736,6 +1736,27 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  /**
+   * The text track changed event object
+   * @param {FakeEvent} event - payload with text track
+   * @returns {void}
+   * @private
+   */
+  _onTextTrackAdded(event: FakeEvent): void {
+    const videoElement = this.getVideoElement();
```

Comment:
we want to avoid accessing the native video element as much as possible, for example, this will fail for flash and youtube or any other external non HTML5 engine.

Reply:
@OrenMe  why? get video element will be null and then nothing happens on the next check, otherwise, text tracks will be empty and if both are not empty for other engines it's ok that the logic will work

Reply:
Ok, but this logic was not needed till now for non - native text track, so what has changed?
We should strive to get away from this and not add more use cases.

Reply:
Flash and youtube won't fire this event anyway

Reply:
it's for native text track only, listener added only for `useNativeTextTrack=true`